src/session/content/message_row/sticker.rs

```diff
@@ -48,48 +47,52 @@ mod imp {
 
 glib::wrapper! {
     pub struct MessageSticker(ObjectSubclass<imp::MessageSticker>)
-        @extends gtk::Widget;
-}
-
-impl Default for MessageSticker {
-    fn default() -> Self {
-        Self::new()
-    }
+        @extends gtk::Widget, MessageRow;
 }
 
 impl MessageSticker {
-    pub fn new() -> Self {
-        glib::Object::new(&[]).expect("Failed to create MessageSticker")
+    pub fn new(message: &glib::Object) -> Self {
+        glib::Object::new(&[("message", message)]).expect("Failed to create MessageSticker")
     }
 
-    pub fn set_message(&self, message: &Message) {
-        if let MessageContent::MessageSticker(data) = message.content().0 {
-            let self_ = imp::MessageSticker::from_instance(self);
-            self_
-                .paintable
-                .set_aspect_ratio(data.sticker.width as f64 / data.sticker.height as f64);
-
-            if data.sticker.sticker.local.is_downloading_completed {
-                self.load_sticker(&data.sticker.sticker.local.path);
+    fn update_widget(&self) {
+        if let Some(message) = self.message() {
+            let (content, session) = if let Some(message) = message.downcast_ref::<Message>() {
+                (message.content().0, message.chat().session())
+            } else if let Some(sponsored_message) = message.downcast_ref::<SponsoredMessage>() {
```

Comment:
I just got contacted by a tdlib developer that only the sponsored messages can only have `MessageText` as content type for now, so I think this can be dropped for now.